### PR TITLE
Update dsfr

### PR DESCRIPTION
**Pull request by nicolaskempf57**
- update DSFR to 1.2.1
- update dependencies to fix critical vulnerabilities
- update nvm to new LTS v16.13.0 (minimal update is v14.18.0 for DSFR)

Close https://github.com/etalab/data.gouv.fr/issues/618

### CODE REVIEW

File: theme/less/override.less [hunk 1]
```diff
@@ -11,6 +11,7 @@ ul {
 
 a:not([href]) {
     cursor: pointer;
+    color: unset;
```
Comment (maudetes) [inline theme/less/override.less:14]: Where is it needed?

**Review by maudetes — COMMENTED** [review]
Nice ! :)

I'm mostly asking for precision on some changes I couldn't understand.

Additionally, I think the grey background add some accessibility issues? For example on the Summary, with blue links, we went from 4.10 to 3.76 contrast score (using the accessibility module on firefox).
I guess we shouldn't deploy this PR as long as we haven't changed the rest of the colors accordingly? Could we make the change on the beige in the colors PR instead ?

File: theme/less/components/read-more.less [hunk 2]
```diff
@@ -3,8 +3,10 @@
     overflow: hidden;
 }
 
-.read-more-link {
+a.read-more-link {
     --gradient-color: var(--w);
+    --blend-size: 100%;
```
[the remark this answers is not in the document]
Comment (nicolaskempf57) [reply theme/less/components/read-more.less:8]: Links doesn't have any background in DSFR so they use this `blend-size` to make a transition on hover from 0% to 100%. But in our case, we want a full size background even without hover to display our transparent gradient.

[the remark this answers is not in the document]
Comment (nicolaskempf57) [reply theme/less/components/read-more.less:8]: Links doesn't have any background in DSFR so they use this `blend-size` to make a transition on hover from 0% to 100%. But in our case, we want a full size background even without hover to display our transparent gradient.

File: theme/less/content/typography.less [hunk 3]
```diff
@@ -147,6 +147,7 @@ h6,
     margin-top: 0;
     font-weight: bold;
     line-height: @line-height-base;
+    color: inherit;
```
[the remark this answers is not in the document]
Comment (nicolaskempf57) [reply theme/less/content/typography.less:150]: Title have a default color of black from the DSFR so we revert it with this inherit. It allow us to have a `.text-white` for the participate section and its title with the correct color.

File: theme/less/override.less (hunk 1 above)
Comment (nicolaskempf57) [reply theme/less/override.less:14]: The DSFR add a new default color to link without href.

In udata-front, links used with @click doesn't have a href. This is bad for accessibility reason so I remove them  when I see them but it's a pattern in our codebase.

**Comment by nicolaskempf57** [comment]
> Nice ! :)
> 
> I'm mostly asking for precision on some changes I couldn't understand.

Most changes reflect DSFR changes to revert them so it's not really easy to understand them :/

> 
> Additionally, I think the grey background add some accessibility issues? For example on the Summary, with blue links, we went from 4.10 to 3.76 contrast score (using the accessibility module on firefox). I guess we shouldn't deploy this PR as long as we haven't changed the rest of the colors accordingly? Could we make the change on the beige in the colors PR instead ?

Yes, the grey is darker so it's worst as long as we stay with our blue . We can add back --beige and put it as summary background before https://github.com/etalab/data.gouv.fr/issues/584

**Comment by nicolaskempf57** [comment]
I merged the discussions updates (from master) so I'll do more manuel tests to see if new changes are necessary.

**Review by geoffreyaldebert — APPROVED** [review]
No remarks. By seeing the history of the comments, I temporarily take role of @maudetes and approve this PR during this vacation time.

**Pull request merged**